### PR TITLE
🪟 🎨 🧹 Migrate Card component to SCSS and refactor

### LINKED ISSUE #15790: Cleanup Card components in webapp
Update `Card`, `ContentCard`, `ConnectorCard` components:

* Migrate to SCSS
* Add storybooks if missing
* Add unit tests where necessary (for testing behavior, not how it looks)

### PR DESCRIPTION
**Pull request by natalyjazzviolin**
## What

This closes #15790 and should not alter anything visually in the UI.

1. This refactors the `Card` component to:
* Use SCSS instead of styled components
* Accept a 'title' prop

2. Refactors the `WorkspaceItem` component at `/workspaces` to use an HTML button for better accessibility.

3. Removes the `ContentCard` component and replaces all of its instances with `Card`

4. Updates styling to use SCSS in some components that use `Card` where styled components would have no longer worked.

## How
Refactors `Card` and replaces styled components with SCSS.

## Recommended reading order

**The `Card` component and its styling:**
1. `airbyte-webapp/src/components/base/Card/Card.tsx`
2. `airbyte-webapp/src/components/base/Card/Card.module.scss`

**Where `Card` is replaced with a button at /workspaces:**

3. `airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.tsx`
4. `airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss`

**Where `Card` replaces `ContentCard`:**

5. `airbyte-webapp/src/components/CenteredPageComponents/PaddedCard.tsx`
6. `airbyte-webapp/src/components/CreateConnectionContent/CreateConnectionContent.tsx`
7. `airbyte-webapp/src/components/DeleteBlock/DeleteBlock.tsx`
8. `airbyte-webapp/src/components/Modal/Modal.tsx`
9. `airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/CreditsUsage.tsx`
10. `airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspacesControl.tsx`
11. `airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StateBlock.tsx`
12. `airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StatusView.tsx`
13. `airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/TransformationView.tsx`
14. `airbyte-webapp/src/pages/ConnectionPage/pages/CreationFormPage/components/ExistingEntityForm.tsx`
15. `airbyte-webapp/src/pages/OnboardingPage/components/UseCaseBlock.tsx`
16. `airbyte-webapp/src/pages/SettingsPage/pages/ConfigurationsPage/ConfigurationsPage.tsx`
17. `airbyte-webapp/src/pages/SettingsPage/pages/SettingsComponents.tsx`
18. `airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx`
19. `airbyte-webapp/src/views/Connector/ServiceForm/index.stories.tsx`

**Where `Card` replaces `ContentCard` and adds SCSS:**

20. `airbyte-webapp/src/views/Connection/CollapsibleCard.module.scss`
21. `airbyte-webapp/src/views/Connection/CollapsibleCard.tsx`
22. `airbyte-webapp/src/components/ConnectionBlock/ConnectionBlock.module.scss`
23. `airbyte-webapp/src/components/ConnectionBlock/ConnectionBlock.tsx`

**Deletes `ContentCard`:**

24. `airbyte-webapp/src/components/ContentCard/ContentCard.tsx`
25. `airbyte-webapp/src/components/ContentCard/index.stories.tsx`
26. `airbyte-webapp/src/components/ContentCard/index.tsx`
27. `airbyte-webapp/src/components/index.tsx`

## 🚨 User Impact 🚨
No breaking changes.

## Tests

<summary><strong>Unit</strong></summary>

None needed.

### CODE REVIEW

File: airbyte-webapp/src/components/base/Card/Card.tsx [hunk 1]
```diff
@@ -1,10 +1,48 @@
...
+  roundedBottom,
+}) => {
+  return (
+    <div
+      className={classNames(
+        className,
+        styles.container,
+        fullWidth ? styles.fullWidth : undefined,
+        withPadding ? styles.withPadding : undefined
+      )}
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/base/Card/Card.tsx:32]:
```suggestion
      className={classNames(className, styles.container, {
        [styles.fullWidth]: fullWidth,
        [styles.withPadding]: withPadding,
      })}
```

File: airbyte-webapp/src/components/base/Card/Card.tsx [hunk 2]
```diff
@@ -1,10 +1,48 @@
...
+    >
+      {title ? (
+        <H5
+          className={classNames(
+            styles.title,
+            lightPadding || !children ? styles.lightPadding : undefined,
+            roundedBottom ? styles.roundedBottom : undefined
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/base/Card/Card.tsx:39]: Repeat above object pattern here 👍

File: airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  width: 100%;
+  background: $white;
+  border-radius: 10px;
+  box-shadow: 0 2px 4px $shadowColor;
+  border: none;
+}
+
+.iconColor {
+  color: #615eff;
+}
```
Comment (krishnaglick) [inline airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss:20]: All these free-standing colors make me a touch nervous. Are we sure these colors can't come from `_colors.scss`?

**Review by krishnaglick — COMMENTED** [review]
I was testing this change on the Connection Status view and it seems to have affected how the Error logs expand. Notably it appears to render the text, then do the element expansion.

File: airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss (hunk 3 above)
Comment (natalyjazzviolin) [reply airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss:20]: Oh I might have missed this one, sorry! I was having trouble importing `_colors.scss` in a few places, will go back and check.

**Comment by natalyjazzviolin** [comment]
> I was testing this change on the Connection Status view and it seems to have affected how the Error logs expand. Notably it appears to render the text, then do the element expansion.

Could you please point me to where this is? I checked out error logs at `/workspaces/<id>/connections/<id>/status` and don't see a difference from the `master` branch
Screen Shot 2022-09-09 at 9 46 37 AM

File: airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss (hunk 3 above)
Comment (krishnaglick) [reply airbyte-webapp/src/packages/cloud/views/workspaces/WorkspacesPage/components/WorkspaceItem.module.scss:19]: If you continue to have issues with it, feel free to shoot me a ping! `@use` is a little weird at first :)

**Comment by krishnaglick** [comment]
> > I was testing this change on the Connection Status view and it seems to have affected how the Error logs expand. Notably it appears to render the text, then do the element expansion.
> 
> Could you please point me to where this is? I checked out error logs at `/workspaces/<id>/connections/<id>/status` and don't see a difference from the `master` branch Screen Shot 2022-09-09 at 9 46 37 AM

https://www.loom.com/share/c265b9bf1d8d40ceb69dc56eb7f34256
It's rendering differently.

**Comment by timroes** [comment]
@krishnaglick I've looked at the Loom with Nataly and also tested master vs this branch. I don't think there is any difference in rendering? The text in both cases will fade to opacity of the content at the same time while expanding it having that short moment where it overlaps the elements below. But I see that as well on master as well as in your Loom recording from master? I think it will just feel slightly different depending on the amount of elements below that it might need to push while animating, why it might feel in your recording from this branch slightly slower, but I don't even feel on the recording that it's slightly slower. If you believe there is an issue on this branch that's not master, could you maybe jump on a call with us, trying to demo it?

**Comment by timroes** [comment]
@krishnaglick No longer relevant. We found a nice way to fix this. Nataly will create a separate PR for this, since it's already present on master and unrelated to this change, so we don't clutter this PR with it.

**Comment by natalyjazzviolin** [comment]
@krishnaglick we also looked at the failing E2E tests - looks like there are no frontend errors and the test is flaky!

**Review by krishnaglick — APPROVED** [review]

**Comment by pmossman** [comment]
@natalyjazzviolin @krishnaglick fyi I think the Frontend: Build is broken here, I can't build airbyte locally off of latest master

**Comment by tealjulia** [comment]
I have a pr up to fix

**Pull request merged**